### PR TITLE
retry-mechanism

### PR DESCRIPTION
**Pull request by KamilForemny**
Retry mechanism if request fails and default retries mechanism form okHttp client.

### CODE REVIEW

File: docker-compose.yaml [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    volumes:
+      - postgres_data:/var/lib/postgresql/data
+    environment:
+      POSTGRES_DB: keycloak
+      POSTGRES_USER: LFY_POSTGRES_KEYCLOAK_USER
+      POSTGRES_PASSWORD: "SajCasdk]=V"
+  keycloak:
+    image: jboss/keycloak:12.0.4
+    volumes:
+      - /Users/kamil/IdeaProjects/keycloak-theme/theme/lottify-theme:/opt/jboss/keycloak/themes/lottify-theme
```
Comment (tholst) [inline docker-compose.yaml]: I think file shouldn't be checked in, or at least not in this form with references to "LFY" and including the actual passwords we use for local development. What do you think?

File: src/main/java/com/thohol/keycloak/JsonGraphQlRemoteClaim.java [hunk 2]
```diff
@@ -38,19 +41,12 @@
     private final static String REMOTE_GRAPHQL = "remote.graphql";
     private final static String REMOTE_GRAPHQL_QUERY = "remote.graphql.query";
     private final static String REMOTE_GRAPHQL_RESULT_PATH = "remote.graphql.path";
-
-
+    private final static String RETRY_REQUEST = "retry.request";
```
Comment (tholst) [inline src/main/java/com/thohol/keycloak/JsonGraphQlRemoteClaim.java]: not really important but `"remote.request.retry"` would be more consistent

```suggestion
    private final static String RETRY_REQUEST = "remote.request.retry";
```

File: src/main/java/com/thohol/keycloak/JsonGraphQlRemoteClaim.java [hunk 3]
```diff
@@ -186,6 +182,15 @@
...
         property.setHelpText("The JSON path of the result data that should be assigned to the custom claim.");
         configProperties.add(property);
+
+        // Retry mechanism
+        property = new ProviderConfigProperty();
+        property.setName(RETRY_REQUEST);
+        property.setLabel("Retry Request");
+        property.setType(ProviderConfigProperty.BOOLEAN_TYPE);
+        property.setDefaultValue("false");
+        property.setHelpText("Retry request when it fails");
```
Comment (tholst) [inline src/main/java/com/thohol/keycloak/JsonGraphQlRemoteClaim.java]:
```suggestion
        property.setHelpText("Retry request when it fails. Will attempt at most two retries.");
```

File: docker-compose.yaml (hunk 1 above)
Comment (KamilForemny) [reply docker-compose.yaml]: ahh yes, I will remove this file

File: src/main/java/com/thohol/keycloak/JsonGraphQlRemoteClaim.java (hunk 3 above)
Comment (KamilForemny) [reply src/main/java/com/thohol/keycloak/JsonGraphQlRemoteClaim.java]: 👍

File: src/main/java/com/thohol/keycloak/HttpHandler.java [hunk 4]
```diff
@@ -23,7 +22,7 @@
             .build();
 
     static JsonNode getJsonNode(boolean retryEnabled, String baseUrl, String contentType, Map<String, String> headers, Map<String, String> queryParameters, Map<String, String> formParameters, String graphQlQuery) {
-        org.apache.log4j.Logger.getLogger(OkHttpClient.class.getName()).setLevel(Level.ALL);
+        LOGGER.info("Stared request, url: " + baseUrl);
```
Comment (tholst) [inline src/main/java/com/thohol/keycloak/HttpHandler.java]: Started?

**Pull request merged**